uk.co.compucorp.civicrm.hrcore/hrcore.php

```diff
@@ -26,7 +26,10 @@ function hrcore_civicrm_config(&$config) {
  * @param array $tasks
  */
 function hrcore_civicrm_searchTasks($objectName, &$tasks) {
-  if ($objectName !== 'contact' || !CRM_Core_Permission::check('create users')) {
+  $isContact = $objectName === 'contact';
```

Comment:
The way this hook implementation is assuming that:
- hrcore only handles search tasks for contacts
- hrcore only handles search tasks that require the "create users" permission

Which is something that doesn't make sense. 

Another problem with it is that we have business logic inside the hook itself, which is a bad practice. Ideally, to avoid having a huge hook implementation in the future, we should put this logic inside a separate function and then call it from within the hook.